nixos/modules/hardware/drivers.nix

```diff
@@ -0,0 +1,135 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.hardware.drivers;
+
+  kernelPackages = config.boot.kernelPackages;
+
+  videoDrivers = config.hardware.gpu.drivers;
+
+  drivers = pkgs.buildEnv {
+    name = "hardware-drivers";
+    paths = cfg.packages;
+  };
+
+  drivers32 = pkgs.buildEnv {
+    name = "hardware-drivers-32bit";
+    paths = cfg.packages32;
+  };
+
+in
+
+{
+
+  imports = [
+    (mkRenamedOptionModule [ "hardware" "opengl" "enable" ] [ "hardware" "drivers" "enable" ])
+    (mkRenamedOptionModule [ "hardware" "opengl" "package" ] [ "hardware" "drivers" "packages" ])
+    (mkRenamedOptionModule [ "hardware" "opengl" "extraPackage" ] [ "hardware" "drivers" "packages" ])
+    (mkRenamedOptionModule [ "hardware" "opengl" "driSupport32Bit" ] [ "hardware" "drivers" "enable32bit" ])
+    (mkRenamedOptionModule [ "services" "xserver" "vaapiDrivers" ] [ "hardware" "drivers" "packages" ])
+    (mkRemovedOptionModule [ "hardware" "opengl" "s3tcSupport" ] ''
+      S3TC support is now always enabled in Mesa.
+    '')
+    (mkRemovedOptionModule [ "hardware" "opengl" "driSupport" ] ''
+      dri support is now always enabled in Mesa.
+    '')
```

Comment:
Please only start warning when the new options are available on all supported releases. To sketch the issue, otherwise, you're putting "3rd party" repos in a spot where they have to behave differently for each version just to avoid user complaints about warnings. This "3rd party" includes `nixos-hardware`.

Reply:
You could use `mkRenamedOptionModuleWith { sinceRelease = 2211; from = ...; to = ...; }`. This will start warning on `master` when 2205 is EOL.

Reply:
didn't realise this existed. It should be used more often, as a lot times I just reference similar modules for correctness